lib/chat_api_web/channels/notification_channel.ex

```diff
@@ -51,21 +46,28 @@ defmodule ChatApiWeb.NotificationChannel do
   end
 
   def handle_in("shout", payload, socket) do
-    {:ok, message} = Messages.create_message(payload)
-    result = ChatApiWeb.MessageView.render("message.json", message: message)
-    # TODO: write doc explaining difference between push, broadcast, etc.
-    push(socket, "shout", result)
+    with %{current_user: current_user} <- socket.assigns,
+         %{id: user_id, account_id: account_id} <- current_user do
+      msg = Map.merge(payload, %{"user_id" => user_id, "account_id" => account_id})
+      {:ok, message} = Messages.create_message(msg)
+      result = ChatApiWeb.MessageView.render("message.json", message: message)
 
-    case result do
-      %{conversation_id: conversation_id} ->
-        topic = "conversation:" <> conversation_id
+      # TODO: write doc explaining difference between push, broadcast, etc.
+      push(socket, "shout", result)
 
-        ChatApiWeb.Endpoint.broadcast_from!(self(), topic, "shout", result)
+      case result do
```

Comment:
Any reason we are using case for a single statement check instead of just an if else?

Reply:
Or are there other states we haven't handled yet and should maybe adds some todos here?

Reply:
hmm i think we probably don't even need the `case` tbh, i think the `result` should always have a `conversation_id` if it reaches this point... maybe we could just do 
```ex
topic = "conversation:" <> result.conversation_id

ChatApiWeb.Endpoint.broadcast_from!(self(), topic, "shout", result)
```
i'll update 👍